### PR TITLE
Core clr2

### PR DESCRIPTION
**Pull request by smudge202**
I almost bricked my windows installing the Win80 sdk so I simply can't do all the projects.

However, if you're happy with the pattern I've gone with here @dennisdoomen this should be very easy to expand upon.

Considerable change; I'm happy to continue working on the components I can and any changes required as a result of PR review.

I'll try to remain very active in the [gitter channel](https://gitter.im/dennisdoomen/fluentassertions) for any questions.

I'll also continue adding remaining profiles (those that I can get SDKs for compatible with Win10!).

### CODE REVIEW

**Comment by smudge202** [comment]
For what it's worth, @mattrigway and @herecydev are generally responsible for reviewing my code, so they may help with general code cleanup.

File: FluentAssertions.Core/TypeInfo.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			this.type = type;
+		}
+
+		public Assembly Assembly { get { return type.Assembly; } }
+
+		public CustomAttributeData[] CustomAttributes
+		{
+			get
+			{
+#if CUSTOM_ATTRIBUTEDATA
```
Comment (herecydev) [inline FluentAssertions.Core/TypeInfo.cs]: If this isn't defined then you won't have a return value for the get. Therefore no compilation.

File: src/FluentAssertions/project.lock.json [hunk 2]
```diff
@@ -0,0 +1,910 @@
+{
```
Comment (herecydev) [inline src/FluentAssertions/project.lock.json]: This shouldn't be included

**Comment by herecydev** [comment]
Looking good to me @smudge202, I haven't checked behavior but I'm sure there are plenty of tests covering this.

**Comment by smudge202** [comment]
Thanks for taking a look @herecydev and nice spots.

I haven't added behaviour so hopefully it makes sense to have not added tests. The intent was to not modify behaviour so simply comply with existing tests. Unfortunately I failed at that (there are 3 failing tests as mentioned in [gitter](https://gitter.im/dennisdoomen/fluentassertions)).

**Comment by dennisdoomen** [comment]
Now that @onovotny has completed #288. Is this now fully covered? I will release 4.0 beta in a couple of days.

**Comment by smudge202** [comment]
Yes, this PR can die now.

**Comment by dennisdoomen** [comment]
OK. Regardless, thank for trying to make this happen.

**Pull request closed without merging**